profile/src/main/java/rdx/works/profile/cloudbackup/domain/CheckBackupStatusUseCase.kt

```diff
@@ -36,31 +32,20 @@ class CheckBackupStatusUseCase @AssistedInject constructor(
             it is ProfileState.NotInitialised
         }.first()
         val profile = (profileState as? ProfileState.Restored)?.profile ?: return Result.success()
-        if (!profile.canBackupToCloud) return Result.success()
 
-        val lastBackupEvent = preferencesManager.lastCloudBackupEvent.firstOrNull()
+        val lastCloudBackupEvent = preferencesManager.lastCloudBackupEvent.firstOrNull()
 
-        return if (lastBackupEvent == null || profile.header.lastModified > lastBackupEvent.profileModifiedTime) {
+        return if (
+            profile.canBackupToCloud &&
+            (lastCloudBackupEvent == null || profile.header.lastModified > lastCloudBackupEvent.profileModifiedTime)
+        ) {
             cloudBackupSyncExecutor.requestCloudBackup()
             Result.success()
         } else {
             Timber.tag("CloudBackup").d("\uD83D\uDEDC Check Backup Status")
-            driveClient.getCloudBackupEntity(fileId = lastBackupEvent.fileId, profile = profile).fold(
-                onSuccess = {
-                    Timber.tag("CloudBackup").d("\uD83D\uDEDC Check Backup Status: All good ✅")
-                    // File still exists, no further action required
-                    cloudBackupErrorStream.resetErrors()
-                    Result.success()
-                },
-                onFailure = { exception ->
-                    Timber.tag("CloudBackup").d(exception, "\uD83D\uDEDC Check Backup Status: Error ❌")
-                    if (exception is BackupServiceException) {
-                        cloudBackupErrorStream.onError(exception)
-                    }
-
-                    Result.success()
-                }
-            )
+            return lastCloudBackupEvent?.let {
```

Comment:
Is this necessary? The same check happens in the use case too.

Reply:
yes it is after the refactoring.

Reply:
previously it was on the top `if (!profile.canBackupToCloud) return Result.success()` 
now it's inside the else